### PR TITLE
Adding roles for logged in user for local development

### PR DESCRIPTION
**Pull request by komalg1**
## Purpose

* ...
Based on the principal id, roles can be assigned to the logged in user for local development
## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?
Adds roles for logged in user with principal type 'User'

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: docs/LOCAL_DEPLOYMENT.md [hunk 1]
```diff
@@ -30,6 +30,30 @@ The files for the dev container are located in `/.devcontainer/` folder.
 
 To customize the accelerator or run it locally, first, copy the `.env.sample` file to your development environment's `.env` file, and edit it according to [environment variable values table](#environment-variables) below.
 
+
+To run the accelerator in local you need to assign some roles to your principal id. You can do it either manually or programatically.
+
+### Manually assign roles
+You need to assign the following roles to your `PRINCIPALID` (you can get your 'principal id' from Microsoft Entra ID):
+1. Cognitive Services OpenAI Contributor role - a001fd3d-188f-4b5d-821b-7da978bf7442
```
Comment (ross-p-smith) [inline docs/LOCAL_DEPLOYMENT.md:38]: How about a table?
```suggestion
| Role | ed |
|----|----|
|  Cognitive Services OpenAI Contributor | a001fd3d-188f-4b5d-821b-7da978bf7442 |
```

File: docs/LOCAL_DEPLOYMENT.md [hunk 2]
```diff
@@ -30,6 +30,30 @@ The files for the dev container are located in `/.devcontainer/` folder.
...
+4. Storage Blob Data Reader role - 2a2b9908-6ea1-4ae2-8e65-a410df84e7d1 
+5. Reader role - acdd72a7-3385-48ef-bd42-f606fba81ae7
+
+### Programatically assign roles
+You can also update the `principalId` value with your own principalId in `deployment.bicep` file.
+Build the bicep file using the below command:
+
+```shell
+cd infra
+bicep build deployment.bicep
```
Comment (ross-p-smith) [inline docs/LOCAL_DEPLOYMENT.md:50]: Why do we need both steps - I thought `az deployment group create` would work with bicep files too

**Review by ross-p-smith — APPROVED** [review]

File: docs/LOCAL_DEPLOYMENT.md (hunk 1 above)
Comment (komalg1) [reply docs/LOCAL_DEPLOYMENT.md]: Updated

File: docs/LOCAL_DEPLOYMENT.md [hunk 3]
```diff
@@ -30,6 +30,30 @@ The files for the dev container are located in `/.devcontainer/` folder.
...
+```
+Then deploy the bicep file using the below command:
+```shell
+az deployment group create --resource-group {your_resource_group} --template-file deployment.json --subscription {your_azure_subscription_id}
```
Comment (ross-p-smith) [inline docs/LOCAL_DEPLOYMENT.md]:
```suggestion
az deployment group create --resource-group {your_resource_group} --template-file deployment.bicep --subscription {your_azure_subscription_id}
```

File: docs/LOCAL_DEPLOYMENT.md [hunk 4]
```diff
@@ -30,6 +30,27 @@ The files for the dev container are located in `/.devcontainer/` folder.
...
 
+
+To run the accelerator in local you need to assign some roles to your principal id. You can do it either manually or programatically.
+
+### Manually assign roles
+You need to assign the following roles to your `PRINCIPALID` (you can get your 'principal id' from Microsoft Entra ID):
+| Role | GUID |
+|----|----|
+|  Cognitive Services OpenAI Contributor | a001fd3d-188f-4b5d-821b-7da978bf7442 |
+| Search Service Contributor role | 7ca78c08-252a-4471-8644-bb5ff32d4ba0
```
Comment (ross-p-smith) [inline docs/LOCAL_DEPLOYMENT.md:41]:
```suggestion
| Search Service Contributor | 7ca78c08-252a-4471-8644-bb5ff32d4ba0
```

File: docs/LOCAL_DEPLOYMENT.md [hunk 5]
```diff
@@ -30,6 +30,27 @@ The files for the dev container are located in `/.devcontainer/` folder.
...
+| Search Index Data Contributor | 8ebe5a00-799e-43f5-93ac-243d3dce84a7
+| Storage Blob Data Reader role | 2a2b9908-6ea1-4ae2-8e65-a410df84e7d1 
+| Reader role | acdd72a7-3385-48ef-bd42-f606fba81ae7
```
Comment (ross-p-smith) [inline docs/LOCAL_DEPLOYMENT.md:44]:
```suggestion
| Reader | acdd72a7-3385-48ef-bd42-f606fba81ae7
```

File: docs/LOCAL_DEPLOYMENT.md (hunk 2 above)
Comment (ross-p-smith) [reply docs/LOCAL_DEPLOYMENT.md]: Create the json file as a pre-commit hook #209

File: docs/LOCAL_DEPLOYMENT.md [hunk 6]
```diff
@@ -30,6 +30,27 @@ The files for the dev container are located in `/.devcontainer/` folder.
...
 
+
+To run the accelerator in local you need to assign some roles to your principal id. You can do it either manually or programatically.
+
+### Manually assign roles
+You need to assign the following roles to your `PRINCIPALID` (you can get your 'principal id' from Microsoft Entra ID):
+| Role | GUID |
+|----|----|
+|  Cognitive Services OpenAI Contributor | a001fd3d-188f-4b5d-821b-7da978bf7442 |
+| Search Service Contributor | 7ca78c08-252a-4471-8644-bb5ff32d4ba0
```
Comment (ross-p-smith) [inline docs/LOCAL_DEPLOYMENT.md:41]:
```suggestion
| Search Service Contributor | 7ca78c08-252a-4471-8644-bb5ff32d4ba0 |
```

File: docs/LOCAL_DEPLOYMENT.md [hunk 7]
```diff
@@ -30,6 +30,27 @@ The files for the dev container are located in `/.devcontainer/` folder.
...
+| Search Index Data Contributor | 8ebe5a00-799e-43f5-93ac-243d3dce84a7
```
Comment (ross-p-smith) [inline docs/LOCAL_DEPLOYMENT.md:42]:
```suggestion
| Search Index Data Contributor | 8ebe5a00-799e-43f5-93ac-243d3dce84a7 |
```

File: docs/LOCAL_DEPLOYMENT.md [hunk 8]
```diff
@@ -30,6 +30,27 @@ The files for the dev container are located in `/.devcontainer/` folder.
...
+| Storage Blob Data Reader | 2a2b9908-6ea1-4ae2-8e65-a410df84e7d1 
+| Reader | acdd72a7-3385-48ef-bd42-f606fba81ae7
```
Comment (ross-p-smith) [inline docs/LOCAL_DEPLOYMENT.md:44]:
```suggestion
| Reader | acdd72a7-3385-48ef-bd42-f606fba81ae7 |
```

**Review by ross-p-smith — APPROVED** [review]

**Pull request merged**